src/visualizers/panels/ExecutionIndex/ExecutionIndexControl.js

```diff
@@ -38,44 +40,97 @@ define([
     };
 
     ExecutionIndexControl.prototype._initWidgetEventHandlers = function () {
-        this._widget.setExecutionDisplayed = this.setExecutionDisplayed.bind(this);
+        this._widget.setDisplayedExecutions = this.setDisplayedExecutions.bind(this);
+    };
+
+    ExecutionIndexControl.prototype.setDisplayedExecutions = function (displayedIds) {
+        this.displayedExecutions = displayedIds;
+        this._updateGraphWidget();
     };
 
-    ExecutionIndexControl.prototype.setExecutionDisplayed = function (id, bool) {
-        var lines = this._linesForExecution[id] || [],
-            otherLines,
-            wasMultiLine = this.displayedExecCount() > 1,
-            isMultiLine;
-
-        this._logger.info(`setting execution ${id} to ${bool ? 'displayed' : 'hidden'}`);
-        this.displayedExecutions[id] = bool;
-
-        // If we just crossed the multi line threshold, then update all the lines
-        isMultiLine = this.displayedExecCount() > 1;
-        if (isMultiLine !== wasMultiLine) {
-            // Refresh the other lines visible
-            otherLines = Object.keys(this.displayedExecutions)
-                .filter(eId => this.displayedExecutions[eId] && (eId !== id))
-                .map(id => this._linesForExecution[id] || [])
-                .reduce((l1, l2) => l1.concat(l2), []);
-
-            this._updateLines(otherLines, false);
-            this._updateLines(otherLines, true);
+    ExecutionIndexControl.prototype._updateGraphWidget = function () {
+        const action = this.displayedExecCount() === 0 ? 'addNode' : 'updateNode';
+        let plotlyJSON = this._consolidateGraphData(this.displayedExecutions);
+        if (plotlyJSON) {
```

Comment:
Is there a relationship between this and the number of displayed executions? It looks like `plotlyJSON` is only falsy if `this.displayedExecutions.length === 0`. In that case, it might be more clear to do something like:
```
if (this.displayedExecCount() > 0) {
    const plotlyJSON = this._consolidateGraphData(this.displayedExecutions);
    this._widget.update(plotlyJSON);
} else {
    this._widget.removeNode();  // Or some equivalent like "clear"
}
```

Reply:
This has been adresed in the latest commit.